### ISSUE TITLE
Rendering into a named outlet breaks subroutes' rendering into their parents

### ISSUE THREAD

**Issue by marcoow** [emberjs/ember.js#9565]
I'm not sure whether I found a bug but I couldn't figure out why what I observe is happening. It looks like as soon as a routes renders into a named outlet its subroutes wouldn't render into their respective parent routes anymore but into the main outlet of the application template.

I created a JSBin that illustrates the problem here: http://emberjs.jsbin.com/yijehigegi/2/edit?html,js

Basically I have 3 nested routes like this:

```js
App.Router.map(function() {
  this.resource('outer', function() {
    this.resource('mid', function() {
      this.route('inner');
    });
  });
});
```

Templates look like this:

```handlebars
//application template
{{outlet}}
{{outlet 'special'}}

//"outer" template
Outer
{{outlet}}

//"mid" template
Mid
{{outlet}}

//"inner" template
Inner
```

Without doing anything else everything works as expected and the "outer" route renders into the application template, the "mid" route renders into the "outer" template and the "inner" route renders into the "mid" template.

As soon as I render the "outer" route into the application template's "special" outlet though:

```js
App.OuterRoute = Ember.Route.extend({
  renderTemplate: function() {
    this.render({
      into:   'application',
      outlet: 'special'
    });
  }
});
```

the "mid" route doesn't render into the "outer" template anymore but into the "main" outlet of the application template (the "inner" route still renders into the "mid" template.)

Maybe that's expected behavior but I couldn't make sense of it.

**Comment by marcoow** [comment]
Any updates on this? Would like to help fix it but I don't think I understand the internals of the router well enough...

**Comment by ef4** [comment]
I'm pretty sure this got fixed by the outlet refactor in 1.11. Please reopen if it can still be reproduced there.

**Comment by marcoow** [comment]
Can verify it works now: http://emberjs.jsbin.com/gotesizave/1/edit?html,js,output

:+1:
